### PR TITLE
Execute message derive macro

### PR DESCRIPTION
**Pull request by CyberHoward**
## :loudspeaker: Type of change

:white_small_square: Bugfix
:white_small_square: New feature
:white_small_square: Enhancement
:white_small_square: Refactoring
:white_small_square: Maintenance

## :scroll: Description and Motivation

A derive macro that can be added to an `ExecuteMsg` struct to generate functions that can be called on the associated boot object. 

Example: 

```rust
#[cw_serde]
#[derive(ExecuteFns)]
pub enum ExecuteMsg{
    /// Freeze will make a mutable contract immutable, must be called by an admin
    Freeze {},
    /// UpdateAdmins will change the admin set of the contract, must be called by an existing admin,
    /// and only works if the contract is mutable
    UpdateAdmins { admins: Vec<String> },
    #[payable]
    Deposit {}
}

#[boot_contract(Empty,ExecuteMsg,Empty,Empty)]
struct Cw1

impl<Chain: BootEnvironment + Clone> Cw1<Chain> {
    pub fn test_macro(&self) {
        self.freeze().unwrap();
        self.update_admins(vec![]).unwrap(); 
        self.deposit(&[Coin::new(13,"juno")]).unwrap();
    }
}
```

## :pencil: Checklist

- Add flag for coins instead of optionals.

:white_small_square: Reviewed submitted code
:white_small_square: Added tests to verify changes
:white_small_square: Updated docs
:white_small_square: Verified on testnet

### CODE REVIEW

File: boot-core/src/lib.rs [hunk 1]
```diff
@@ -45,39 +46,4 @@ pub(crate) mod cosmos_modules {
     pub use cosmrs::proto::cosmos::tx::v1beta1 as tx;
     pub use cosmrs::proto::cosmwasm::wasm::v1 as cosmwasm;
     pub use cosmrs::proto::tendermint::abci as tendermint_abci;
-}
-
-// mod macro_dev {
-//     use terra_rust_script_derive::contract;
-
-//     #[derive(Clone, Debug, contract)]
```
Comment (adairrr) [inline boot-core/src/lib.rs:53]: Is this what the `terra_rust_script_derive::contract` had done before?

File: packages/boot-exec-derive/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
+#![recursion_limit = "128"]
+
+extern crate proc_macro;
+use convert_case::{Case, Casing};
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, DeriveInput, Fields, Ident};
+
+#[proc_macro_derive(ExecuteFns,attributes(payable))]
+pub fn boot_execute(input: TokenStream) -> TokenStream {
```
Comment (adairrr) [inline packages/boot-exec-derive/src/lib.rs:10]: Will this be able to handle more advanced top-level structs like the `Middleware` ones with generics?

File: boot-core/src/lib.rs (hunk 1 above)
Comment (CyberHoward) [reply boot-core/src/lib.rs:53]: Yes and no, the old macro just returned the ExecuteMsg variant. This generates a trait and implements it for any contract with that ExecuteMsg. The `payable` attribute lets you specify which functions should accept a payment. All the generated functions can then be called on any boot-object (with any boot-compatible backend environment)

Comment (CyberHoward) [reply boot-core/src/lib.rs:53]: I've been waiting with writing our testing infrastructure for this haha

File: packages/boot-exec-derive/src/lib.rs (hunk 2 above)
Comment (CyberHoward) [reply packages/boot-exec-derive/src/lib.rs:10]: Nope this is just for regular contracts. We could augment it to fit our needs in the Abstract crates.

File: boot-core/src/lib.rs (hunk 1 above)
Comment (adairrr) [reply boot-core/src/lib.rs:53]: > Yes and no, the old macro just returned the ExecuteMsg variant. This generates a trait and implements it for any contract with that ExecuteMsg. The `payable` attribute lets you specify which functions should accept a payment. All the generated functions can then be called on any boot-object (with any boot-compatible backend environment) 

Yeah the payable attribute seems really versatile and smart contract authors can use it as a reference to add others ~

**Pull request merged**